nimbus-jose-jwt_aws-kms-extension/src/main/java/com/nimbusds/jose/aws/kms/crypto/impl/KmsAsymmetricECDSAProvider.java

```diff
@@ -0,0 +1,114 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package com.nimbusds.jose.aws.kms.crypto.impl;
+
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.model.MessageType;
+import com.amazonaws.services.kms.model.SigningAlgorithmSpec;
+import com.google.common.collect.ImmutableMap;
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.impl.BaseJWSProvider;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.var;
+import org.apache.commons.codec.digest.MessageDigestAlgorithms;
+
+
+/**
+ * This class provides cryptography support for RSA-SSA based signing/verification with keys stored in AWS KMS.
+ */
+public abstract class KmsAsymmetricECDSAProvider extends BaseJWSProvider {
+
+    /**
+     * AWS-KMS client.
+     */
+    @NonNull
+    @Getter(AccessLevel.PROTECTED)
+    private final AWSKMS kms;
+
+    /**
+     * KMS private-key (CMK) ID (it can be a key ID, key ARN, key alias or key alias ARN)
+     */
+    @NonNull
+    @Getter(AccessLevel.PROTECTED)
+    private final String privateKeyId;
+
+    /**
+     * KMS Message Type. Refer KMS's sign and verify APIs for details.
+     * Ref: https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html#KMS-Sign-request-MessageType
+     */
+    @NonNull
+    @Getter(AccessLevel.PROTECTED)
+    private final MessageType messageType;
+
+    public static final Map<JWSAlgorithm, String> JWS_ALGORITHM_TO_MESSAGE_DIGEST_ALGORITHM =
```

Comment:
Looking at this PR, it seems apart from this map, everything else is exactly same as the classes present for KmsAsymmetricRSASSA. It would be great if you could move some common code to helper classes and then can call it from both `KmsAsymmetricRSASSA*` classes and `KmsAsymmetricECDSA*` classes.